tests/rptest/tests/e2e_topic_recovery_test.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2021 Vectorized, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from rptest.services.cluster import cluster
+from ducktape.mark import matrix
+from ducktape.cluster.cluster_spec import ClusterSpec
+from rptest.clients.types import TopicSpec
+from rptest.services.redpanda import RedpandaService, SISettings
+from rptest.util import Scale
+from rptest.clients.rpk import RpkTool
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.services.franz_go_verifiable_services import FranzGoVerifiableProducer, FranzGoVerifiableSeqConsumer
+from rptest.util import wait_until
+import time
+import json
+
+
+class EndToEndTopicRecovery(RedpandaTest):
+    """
+    The test produces data and makes sure that all data
+    is uploaded to S3. After that the test recreates the
+    cluster (or topic) and runs the verifier.
+    """
+
+    topics = (TopicSpec(
+        partition_count=1,
+        replication_factor=3,
+    ), )
+
+    def __init__(self, test_context):
+        si_settings = SISettings(
+            log_segment_size=1024 * 1024,
+            cloud_storage_segment_max_upload_interval_sec=5)
+        self.scale = Scale(test_context)
+        self._bucket = si_settings.cloud_storage_bucket
+        super().__init__(test_context=test_context, si_settings=si_settings)
+        self._ctx = test_context
+        self._producer = None
+        self._consumer = None
+        self._verifier_node = test_context.cluster.alloc(
+            ClusterSpec.simple_linux(1))[0]
+        self.logger.info(f"Verifier node name: {self._verifier_node.name}")
+
+    def init_producer(self, msg_size, num_messages):
+        self._producer = FranzGoVerifiableProducer(self._ctx, self.redpanda,
+                                                   self.topic, msg_size,
+                                                   num_messages,
+                                                   [self._verifier_node])
```

Comment:
nit: shouldn't have to do `[self._verifier_node]`. Passing `self._verifier_node` should work.

Reply:
it won't work because `_verifier_node` is not an array